src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -231,7 +231,7 @@ private function extractFromAccessor($class, $property)
      */
     private function extractFromReflectionType(\ReflectionType $reflectionType)
     {
-        $phpTypeOrClass = (string) $reflectionType;
+        $phpTypeOrClass = method_exists($reflectionType, 'getName') ? $reflectionType->getName() : (string) $reflectionType;
```

Comment:
If there is a `getName` method, then we won't be using `__toString`, and if we are using `__toString` that means it's PHP <= 7.0 so there is no `?` prepended anyway. Stripping of characters is only a temporary solution anyway, as the `__toString` is likely to keep changing, judging from the mailing list discussion.